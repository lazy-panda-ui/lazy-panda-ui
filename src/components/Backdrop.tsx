import React from 'react';
import { View, StyleSheet, ViewStyle } from 'react-native';

export interface BackdropProps {
  open: boolean;
  style?: ViewStyle;
  children?: React.ReactNode;
}

export const Backdrop: React.FC<BackdropProps> = ({ open, style, children }) => {
  if (!open) return null;
  return (
    <View style={[styles.backdrop, style]}>
      {children}
    </View>
  );
};

const styles = StyleSheet.create({
  backdrop: {
    ...StyleSheet.absoluteFillObject,
    backgroundColor: 'rgba(0,0,0,0.5)',
    justifyContent: 'center',
    alignItems: 'center',
    zIndex: 1000,
  },
});
