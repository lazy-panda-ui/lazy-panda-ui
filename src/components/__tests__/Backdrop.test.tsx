import React from 'react';
import { render } from '@testing-library/react-native';
import { Backdrop } from '../Backdrop';

describe('Backdrop', () => {
  it('renders when open', () => {
    const { toJSON } = render(<Backdrop open={true} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
