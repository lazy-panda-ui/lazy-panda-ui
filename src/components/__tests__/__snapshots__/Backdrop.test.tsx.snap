// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop renders when open 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.5)",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 1000,
      },
      undefined,
    ]
  }
/>
`;
